src/zenml/deployers/base_deployer.py

```diff
@@ -0,0 +1,996 @@
+        )
+
+        start_time = time.time()
+        sleep_time = 5
+        while True:
+            deployment_state = DeploymentOperationalState(
+                status=DeploymentStatus.ERROR,
+            )
+            try:
+                deployment_state = self.do_get_deployment(deployment)
+            except DeploymentNotFoundError:
+                deployment_state = DeploymentOperationalState(
+                    status=DeploymentStatus.ABSENT
+                )
+            except DeployerError as e:
+                logger.exception(
+                    f"Failed to get deployment {deployment.name}: {e}"
+                )
+            finally:
+                deployment = self._update_deployment(
+                    deployment, deployment_state
+                )
+
+            if deployment.status in [
+                desired_status,
+                DeploymentStatus.ERROR,
+            ]:
+                break
+
+            elapsed_time = int(time.time() - start_time)
+            if elapsed_time > timeout:
+                raise DeploymentTimeoutError(
+                    f"Timed out waiting for deployment {deployment.name} "
+                    f"to reach desired state '{desired_status}' after {timeout} "
+                    "seconds"
+                )
+            logger.debug(
+                f"The deployment {deployment.name} state is still "
+                f"'{deployment.status}' after {elapsed_time} seconds. Waiting for "
+                f"max {timeout - elapsed_time} seconds..."
+            )
+            time.sleep(sleep_time)
+
+        return deployment, deployment_state
+
+    def _get_deployment_analytics_metadata(
+        self,
+        deployment: "DeploymentResponse",
+        stack: Optional["Stack"] = None,
+    ) -> Dict[str, Any]:
+        """Returns the deployment metadata.
+
+        Args:
+            deployment: The deployment to track.
+            stack: The stack on which the pipeline is deployed.
+
+        Returns:
+            the metadata about the deployment
+        """
+        snapshot = deployment.snapshot
+        stack_metadata = {}
+        if stack:
+            stack_metadata = {
+                component_type.value: component.flavor
+                for component_type, component in stack.components.items()
+            }
+        return {
+            "project_id": deployment.project_id,
+            "store_type": Client().zen_store.type.value,
+            **stack_metadata,
+            "deployment_id": str(deployment.id),
+            "snapshot_id": str(snapshot.id) if snapshot else None,
+            "deployer_id": str(self.id),
+            "deployer_flavor": self.flavor,
+            "deployment_status": deployment.status,
+        }
+
+    def provision_deployment(
+        self,
+        snapshot: PipelineSnapshotResponse,
+        stack: "Stack",
+        deployment_name_or_id: Union[str, UUID],
```

Comment:
As far as I can see, the deployment object doesn't exist at this point yet, so this can really only be the name?

Reply:
nvm, it treats it as an ID when replacing 🤔